pkg/resourcekeeper/cache.go

```diff
@@ -101,3 +101,13 @@ func (cache *resourceCache) get(ctx context.Context, mr v1beta1.ManagedResource)
 	}
 	return entry
 }
+
+func (cache *resourceCache) setNotExist(mr v1beta1.ManagedResource) {
+	key := mr.ResourceKey()
+	entry, cached := cache.m[key]
+	if !cached {
+		entry = &resourceCacheEntry{obj: mr.ToUnstructured(), mr: mr}
+	}
+	entry.exists = false
```

Comment:
why we don't set `deleted=true` here?

Reply:
deleted=true is another situation. deleted=true means the resource should be deleted. exists=false means it does not exist. So exists!=deleted means everything ok, but exists==deleted means either resource should be deleted or state kept.